cmd/explaintest/r/new_character_set_builtin.result

```diff
@@ -286,3 +286,207 @@ select ord(a), ord(b), ord(c) from t;
 ord(a)	ord(b)	ord(c)
 14989485	54992	228
 set @@tidb_enable_vectorized_expression = false;
+drop table if exists t;
+create table t (a char(20) charset utf8mb4, b char(20) charset gbk, c binary(20));
+insert into t values ('一', '一', 0xe4b880);
+insert into t values ('一', '一', 0xd2bb);
+insert into t values ('一', '一', 0xe4ba8c);
+insert into t values ('一', '一', 0xb6fe);
+set @@tidb_enable_vectorized_expression = true;
+select hex(concat(a, c)), hex(concat(b, c)) from t;
+hex(concat(a, c))	hex(concat(b, c))
+E4B880E4B8800000000000000000000000000000000000	D2BBE4B8800000000000000000000000000000000000
+E4B880D2BB000000000000000000000000000000000000	D2BBD2BB000000000000000000000000000000000000
+E4B880E4BA8C0000000000000000000000000000000000	D2BBE4BA8C0000000000000000000000000000000000
+E4B880B6FE000000000000000000000000000000000000	D2BBB6FE000000000000000000000000000000000000
+select hex(concat(a, 0xe4b880)), hex(concat(b, 0xd2bb)) from t;
+hex(concat(a, 0xe4b880))	hex(concat(b, 0xd2bb))
+E4B880E4B880	D2BBD2BB
+E4B880E4B880	D2BBD2BB
+E4B880E4B880	D2BBD2BB
+E4B880E4B880	D2BBD2BB
+select a = 0xe4b880, b = 0xd2bb from t;
+a = 0xe4b880	b = 0xd2bb
+1	1
+1	1
+1	1
+1	1
+select a = c, b = c from t;
+a = c	b = c
+0	0
+0	0
+0	0
+0	0
+select hex(insert(a, 1, 2, 0xe4ba8c)), hex(insert(b, 1, 2, 0xb6fe)) from t;
+hex(insert(a, 1, 2, 0xe4ba8c))	hex(insert(b, 1, 2, 0xb6fe))
+E4BA8C	B6FE
+E4BA8C	B6FE
+E4BA8C	B6FE
+E4BA8C	B6FE
+select hex(insert(a, 1, 2, c)), hex(insert(b, 1, 2, c)) from t;
+hex(insert(a, 1, 2, c))	hex(insert(b, 1, 2, c))
+E4B880000000000000000000000000000000000080	E4B8800000000000000000000000000000000000
+D2BB00000000000000000000000000000000000080	D2BB000000000000000000000000000000000000
+E4BA8C000000000000000000000000000000000080	E4BA8C0000000000000000000000000000000000
+B6FE00000000000000000000000000000000000080	B6FE000000000000000000000000000000000000
+select hex(lpad(a, 5, 0xe4ba8c)), hex(lpad(b, 5, 0xb6fe)) from t;
+hex(lpad(a, 5, 0xe4ba8c))	hex(lpad(b, 5, 0xb6fe))
+E4BA8CE4BA8CE4BA8CE4BA8CE4B880	B6FEB6FEB6FEB6FED2BB
+E4BA8CE4BA8CE4BA8CE4BA8CE4B880	B6FEB6FEB6FEB6FED2BB
+E4BA8CE4BA8CE4BA8CE4BA8CE4B880	B6FEB6FEB6FEB6FED2BB
+E4BA8CE4BA8CE4BA8CE4BA8CE4B880	B6FEB6FEB6FEB6FED2BB
+select hex(lpad(a, 5, c)), hex(lpad(b, 5, c)) from t;
+hex(lpad(a, 5, c))	hex(lpad(b, 5, c))
+E4B8E4B880	E4B880D2BB
+D2BBE4B880	D2BB00D2BB
+E4BAE4B880	E4BA8CD2BB
+B6FEE4B880	B6FE00D2BB
+select hex(rpad(a, 5, 0xe4ba8c)), hex(rpad(b, 5, 0xb6fe)) from t;
```

Comment:
mysql has different results here and they confirmed it is bug https://bugs.mysql.com/bug.php?id=105668&thanks=4

Reply:
Please make a note of it and put it in our discussion document.